src/cli.js

```diff
@@ -0,0 +1,1601 @@
+		this.version = version;
+		this.logger.setBanner({
+			name: this.name,
+			copyright: this.copyright,
+			version: this.version
+		});
+		this.debugLogger.timestampEnabled(true);
+
+		process.on('exit', () => {
+			this.debugLogger.trace(`Total run time ${process.uptime().toFixed(2)}s`);
+		});
+	}
+
+	/**
+	 * This method is called by the SDK and we need to keep it.
+	 *
+	 * @access public
+	 * @deprecated
+	 */
+	addAnalyticsEvent() {
+		// noop
+	}
+
+	/**
+	 * Alias for `on()`. This method has been deprecated for years, yet it is
+	 * still used, so we must keep it.
+	 *
+	 * @access public
+	 * @deprecated
+	 */
+	addHook(...args) {
+		return this.on(...args);
+	}
+
+	/**
+	 * Applies commander's argv for current command and all parent command
+	 * contexts into this CLI's argv.
+	 *
+	 * @param {Command} - A Commander.js command context to get parsed options from.
+	 * @access private
+	 */
+	applyArgv(cmd) {
+		if (Array.isArray(cmd?.options)) {
+			const argv = this.argv;
+			const cargv = cmd.opts();
+			this.debugLogger.trace(`Copying ${cmd.options.length} options... (${cmd.name()})`);
+			for (const o of cmd.options) {
+				let name = o.name();
+				if (o.negate) {
+					name = name.replace(/^no-/, '');
+				}
+				this.debugLogger.trace(`  Setting ${name} = ${cargv[o.attributeName()]} (prev: ${argv[name]})`);
+				argv[name] = cargv[o.attributeName()];
+			}
+		}
+	}
+
+	/**
+	 * Adds the config flags, options, arguments, and subcommands to a command.
+	 *
+	 * @param {String} cmdName - The name of the command.
+	 * @param {Command} cmd - The commander command.
+	 * @param {Object} conf - The command configuration.
+	 * @access private
+	 */
+	applyConfig(cmdName, cmd, conf) {
+		if (conf.skipBanner) {
+			this.logger.skipBanner(true);
+		}
+
+		this.command.conf = conf;
+
+		applyCommandConfig(this, cmdName, cmd, conf);
+
+		if (conf.platforms) {
+			this.debugLogger.trace(`Detected conf.platforms applying config for "${cmd.name()}", overriding createHelp()`);
+			this.command.createHelp = () => {
+				return Object.assign(new TiHelp(this, conf.platforms), this.command.configureHelp());
+			};
+		}
+	}
+
+	/**
+	 * Defines a hook function that will emit an event before and after the hooked function is
+	 * invoked.
+	 *
+	 * @param {String} name - The name of hook event.
+	 * @param {Object} [ctx] - The `this` context to bind the callbacks to.
+	 * @param {Function} [fn] - The function being hooked.
+	 * @returns {Function}
+	 * @access public
+	 */
+	createHook(name, ctx, fn) {
+		let dataPayload;
+
+		if (typeof ctx === 'function') {
+			fn = ctx;
+			ctx = null;
+		} else if (ctx && typeof ctx === 'object' && !fn) {
+			dataPayload = ctx;
+			ctx = null;
+		}
+
+		const hookType = typeof fn === 'function' ? 'function' : 'event';
+		this.debugLogger.trace(`Creating ${hookType} hook "${name}"`);
+
+		return (...args) => {
+			let data = Object.assign(dataPayload || {}, {
```

Comment:
Any reason why not to use `??` ?

Reply:
You're not wrong, but I've spent more time fixing bugs in this hook code than anywhere else. I don't want to touch this code if I don't have to.